build_tools/benchmarks/benchmark_helper.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Miscellaneous tool to help work with benchmark suite and benchmark CI."""
+
+import pathlib
+import sys
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+import argparse
+import json
+from typing import List
+
+from e2e_test_framework import serialization
+from e2e_test_artifacts import model_artifacts, iree_artifacts
+from e2e_test_framework.definitions import iree_definitions
+
+
+def dump_flags_of_generation_config(
+    module_generation_config: iree_definitions.ModuleGenerationConfig,
+    root_path: pathlib.PurePath = pathlib.PurePath()):
+
+  imported_model = module_generation_config.imported_model
+  imported_model_path = iree_artifacts.get_imported_model_path(
+      imported_model=imported_model, root_path=root_path)
+  source_model_path = model_artifacts.get_model_path(model=imported_model.model,
+                                                     root_path=root_path)
+
+  compile_flags = module_generation_config.materialize_compile_flags() + [
+      str(imported_model_path)
+  ]
```

Comment:
Some thoughts of the path argument here: It looks like we can have the input file path as part of the `module_generation_config.materialize_compile_flags()`.

The problem is that today we have CMake rules taking the model path as a separate field because internally they need to build the deps on the input file. So it might not be totally suitable to consider the input file as part of the "compile flags"

Reply:
Whether the model flag is part of the compilation flags kind of depends on what you're using it for. If the flags are being produced to feed into some invocation you're about to make, then it does make sense to consider it a flag. But because the path isn't consistent for anything you want to be portable, I don't think it should be. This is already being output in a format that isn't copy-pastable, so in that case I would list the model file separately

Reply:
Makes sense. Ack